### ISSUE TITLE
build failed in Android

### ISSUE THREAD

**Issue by MoeMamdouh** [invertase/react-native-firebase#1107]
### Issue
I follow the instructions to link react-native-firebase in react native app iOS work successfully but android is failed build
https://rnfirebase.io/docs/v4.2.x/installation/android

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Could not find com.google.android.gms:play-services-base:15.0.0.
  Searched in the following locations:
      file:/Users/moe/Library/Android/sdk/extras/m2repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.pom
      file:/Users/moe/Library/Android/sdk/extras/m2repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.jar
      file:/Users/moe/Library/Android/sdk/extras/google/m2repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.pom
      file:/Users/moe/Library/Android/sdk/extras/google/m2repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.jar
      file:/Users/moe/Library/Android/sdk/extras/android/m2repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.pom
      file:/Users/moe/Library/Android/sdk/extras/android/m2repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.jar
      file:/Users/moe/.m2/repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.pom
      file:/Users/moe/.m2/repository/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.jar
      https://jcenter.bintray.com/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.pom
      https://jcenter.bintray.com/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.jar
      file:/Users/moe/Projects/react-native/firebaseTest/node_modules/react-native/android/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.pom
      file:/Users/moe/Projects/react-native/firebaseTest/node_modules/react-native/android/com/google/android/gms/play-services-base/15.0.0/play-services-base-15.0.0.jar
  Required by:
      project :app
> Could not find com.google.firebase:firebase-core:15.0.2.
  Searched in the following locations:
      file:/Users/moe/Library/Android/sdk/extras/m2repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.pom
      file:/Users/moe/Library/Android/sdk/extras/m2repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.jar
      file:/Users/moe/Library/Android/sdk/extras/google/m2repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.pom
      file:/Users/moe/Library/Android/sdk/extras/google/m2repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.jar
      file:/Users/moe/Library/Android/sdk/extras/android/m2repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.pom
      file:/Users/moe/Library/Android/sdk/extras/android/m2repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.jar
      file:/Users/moe/.m2/repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.pom
      file:/Users/moe/.m2/repository/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.jar
      https://jcenter.bintray.com/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.pom
      https://jcenter.bintray.com/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.jar
      file:/Users/moe/Projects/react-native/firebaseTest/node_modules/react-native/android/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.pom
      file:/Users/moe/Projects/react-native/firebaseTest/node_modules/react-native/android/com/google/firebase/firebase-core/15.0.2/firebase-core-15.0.2.jar
  Required by:
      project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s
3 actionable tasks: 1 executed, 2 up-to-date
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/getting-started.html

### Environment

1. Application Target Platform:
Android

2. Development Operating System:
macOS high Sierra 

3. Build Tools:

4. React Native version:
0.55.4

5. RNFirebase Version:
^4.2.0

6. Firebase Module:
 N/A

**Comment by IljaDaderko** [comment]
Try updating play services to 15.0.1

**Comment by MoeMamdouh** [comment]
its already updated @IljaDaderko

**Comment by MoeMamdouh** [comment]
@chrisbianca can you help please?

**Comment by husnaingoldev** [comment]
try cleaning the project

**Comment by chrisbianca** [comment]
It looks like you haven't correctly added `google()` to your build.gradle.  Please check all the installation instructions on the docs.

**Comment by MoeMamdouh** [comment]
@chrisbianca  no I added it, and i follow the instructions and i checked it serval times.
and this is the files

**build.gradle:**
```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    repositories {
       mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        // -------------------------------------------------
        // Add this below the existing maven property above
        // -------------------------------------------------
        google() 
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.2'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
        classpath 'com.google.gms:google-services:3.2.1'
    }
}

allprojects {
    repositories {
        mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}

```

**android/app/build.gradle**

```
apply plugin: "com.android.application"

import com.android.build.OutputFile

/**
 * The react.gradle file registers a task for each build variant (e.g. bundleDebugJsAndAssets
 * and bundleReleaseJsAndAssets).
 * These basically call `react-native bundle` with the correct arguments during the Android build
 * cycle. By default, bundleDebugJsAndAssets is skipped, as in debug/dev mode we prefer to load the
 * bundle directly from the development server. Below you can see all the possible configurations
 * and their defaults. If you decide to add a configuration block, make sure to add it before the
 * `apply from: "../../node_modules/react-native/react.gradle"` line.
 *
 * project.ext.react = [
 *   // the name of the generated asset file containing your JS bundle
 *   bundleAssetName: "index.android.bundle",
 *
 *   // the entry file for bundle generation
 *   entryFile: "index.android.js",
 *
 *   // whether to bundle JS and assets in debug mode
 *   bundleInDebug: false,
 *
 *   // whether to bundle JS and assets in release mode
 *   bundleInRelease: true,
 *
 *   // whether to bundle JS and assets in another build variant (if configured).
 *   // See http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Build-Variants
 *   // The configuration property can be in the following formats
 *   //         'bundleIn${productFlavor}${buildType}'
 *   //         'bundleIn${buildType}'
 *   // bundleInFreeDebug: true,
 *   // bundleInPaidRelease: true,
 *   // bundleInBeta: true,
 *
 *   // whether to disable dev mode in custom build variants (by default only disabled in release)
 *   // for example: to disable dev mode in the staging build type (if configured)
 *   devDisabledInStaging: true,
 *   // The configuration property can be in the following formats
 *   //         'devDisabledIn${productFlavor}${buildType}'
 *   //         'devDisabledIn${buildType}'
 *
 *   // the root of your project, i.e. where "package.json" lives
 *   root: "../../",
 *
 *   // where to put the JS bundle asset in debug mode
 *   jsBundleDirDebug: "$buildDir/intermediates/assets/debug",
 *
 *   // where to put the JS bundle asset in release mode
 *   jsBundleDirRelease: "$buildDir/intermediates/assets/release",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in debug mode
 *   resourcesDirDebug: "$buildDir/intermediates/res/merged/debug",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in release mode
 *   resourcesDirRelease: "$buildDir/intermediates/res/merged/release",
 *
 *   // by default the gradle tasks are skipped if none of the JS files or assets change; this means
 *   // that we don't look at files in android/ or ios/ to determine whether the tasks are up to
 *   // date; if you have any other folders that you want to ignore for performance reasons (gradle
 *   // indexes the entire tree), add them here. Alternatively, if you have JS files in android/
 *   // for example, you might want to remove it from here.
 *   inputExcludes: ["android/**", "ios/**"],
 *
 *   // override which node gets called and with what additional arguments
 *   nodeExecutableAndArgs: ["node"],
 *
 *   // supply additional arguments to the packager
 *   extraPackagerArgs: []
 * ]
 */

project.ext.react = [
    entryFile: "index.js"
]

apply from: "../../node_modules/react-native/react.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion 23
    buildToolsVersion "23.0.1"

    defaultConfig {
        applicationId "com.firebasetest"
        minSdkVersion 16
        targetSdkVersion 22
        versionCode 1
        versionName "1.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation project(':react-native-firebase')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:23.0.1"
    implementation "com.facebook.react:react-native:+"  // From node_modules

    // Firebase dependencies
    implementation "com.google.android.gms:play-services-base:15.0.0"
    implementation "com.google.firebase:firebase-core:15.0.2"
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply plugin: 'com.google.gms.google-services'
```

**android/gradle/wrapper/gradle-wrapper.properties**

```
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```

any help please.

**Comment by IljaDaderko** [comment]
@MoeMamdouh 

You also need to add `google()` dependency inside build.gradle under allProjects

```
allprojects {
    repositories {
        jcenter()
        google()
        maven {
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}
```

**Comment by MoeMamdouh** [comment]
@IljaDaderko 
thank you, yes you're right, also i update **android/app/build.gradle** to
```
compileSdkVersion 27
buildToolsVersion "27.0.3"
```

**Comment by batmanhit** [comment]
These 2 solutions above worked like a charm.
Thanks.

**Comment by MoeMamdouh** [comment]
@batmanhit U R welcome

**Comment by mouthzipper** [comment]
saved my day. Thanks @MoeMamdouh @IljaDaderko

**Comment by MacKentoch** [comment]
Does not solve still having `The library com.google.android.gms:play-services-basement is being requested by various other libraries at [[15.0.0,15.0.0], [15.0.1,15.0.1]], but resolves to 15.0.1.`.

I don't explicitly use play-services-basement.

This looking like related to [firebase release note](https://firebase.google.com/support/release-notes/android#20180523)

**Comment by hegelstad** [comment]
@MacKentoch are you using react-native-maps? In `node_modules/react-native-maps/lib/android/build.gradle` set base and maps version to `15.0.1`.
